lib/semantics/resolve-names.cc

```diff
@@ -293,12 +294,12 @@ class ArraySpecVisitor {
 class ScopeHandler : public virtual ImplicitRulesVisitor {
 public:
   void set_rootScope(Scope &scope) { PushScope(scope); }
-  Scope &CurrScope() { return *scopes_.top(); }
+  Scope &currScope() { return *currScope_; }
   // Return the enclosing scope not corresponding to a derived type:
   Scope &CurrNonTypeScope();
 
   // Create a new scope and push it on the scope stack.
-  Scope &PushScope(Scope::Kind kind, Symbol *symbol);
+  void PushScope(Scope::Kind kind, Symbol *symbol);
```

Comment:
Was the returned reference never used?  Or did it become infeasible to produce?

One of the nice things about functions that return references is that their users know that they can't fail, since there's nothing like a null pointer for references.

Reply:
It wasn't necessary to return it. The caller can get the current scope by calling currScope() like is done everywhere else.